lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -175,14 +175,10 @@ private void deleteGen2Directory(AzureLocation location)
 
     private void deleteBlobDirectory(AzureLocation location)
     {
-        String path = location.path();
-        if (!path.isEmpty() && !path.endsWith("/")) {
-            path += "/";
-        }
         BlobContainerClient blobContainerClient = createBlobContainerClient(location);
-        PagedIterable<BlobItem> blobItems = blobContainerClient.listBlobs(new ListBlobsOptions().setPrefix(path), null);
+        PagedIterable<BlobItem> blobItems = blobContainerClient.listBlobs(new ListBlobsOptions().setPrefix(location.directoryPath()), null);
         for (BlobItem item : blobItems) {
-            String blobUrl = Utility.urlEncode(item.getName());
+            String blobUrl = urlEncode(item.getName());
```

Comment:
Can we check this change via `io.trino.filesystem.AbstractTestTrinoFileSystem#testDeleteDirectory(boolean)` ?

Reply:
I don't follow. This is a good potential regression use-case as well.

Reply:
I was not able to write a test that would fail before adding `urlEncode` in this class.